config/gcos.words

```diff
@@ -0,0 +1,477 @@
+reserved:	B-OR
+reserved:	B-XOR
+reserved:	BEFORE
+reserved:	BINARY
+reserved:	BINARY-LONG
+reserved:	BINARY-SHORT
+reserved:	BIT
+reserved:	BLANK
+reserved:	BLOCK
+reserved:	BOOLEAN
+reserved:	BOTTOM
+reserved:	BSN
+reserved:	BY
+reserved:	CALL
+reserved:	CANCEL
+reserved:	CD
+reserved:	CF
+reserved:	CH
+reserved:	CHARACTER
+reserved:	CHARACTERS
+reserved:	CLASS
+reserved:	CLOCK-UNITS
+reserved:	CLOSE
+reserved:	COBOL
+reserved:	CODE
+reserved:	CODE-SET
+reserved:	COLLATING
+reserved:	COLUMN
+reserved:	COMMA
+reserved:	COMMON
+reserved:	COMMUNICATION
+reserved:	COMP=COMP-3
+reserved:	COMP-1=BINARY-SHORT
+reserved:	COMP-2=BINARY-LONG
+reserved:	COMP-3
+reserved:	COMP-5=PACKED-DECIMAL
+reserved:	COMP-8=PACKED-DECIMAL
+reserved:	COMP-9=FLOAT-SHORT
+reserved:	COMP-10=FLOAT-LONG
+reserved:	COMP-15=FLOAT-LONG
+reserved:	COMPUTATIONAL=COMP-3
+reserved:	COMPUTATIONAL-1=BINARY-SHORT
+reserved:	COMPUTATIONAL-2=BINARY-LONG
+reserved:	COMPUTATIONAL-3
+reserved:	COMPUTATIONAL-5=PACKED-DECIMAL
+reserved:	COMPUTATIONAL-8=PACKED-DECIMAL
+reserved:	COMPUTATIONAL-9=FLOAT-SHORT
+reserved:	COMPUTATIONAL-10=FLOAT-LONG
+reserved:	COMPUTATIONAL-15=FLOAT-LONG
+reserved:	COMPUTE
+reserved:	CONFIGURATION
+reserved:	CONSTANT		# COBOL 2002 != GCOS (unsupported)
+reserved:	CONTAINS
+reserved:	CONTENT
+reserved:	CONTINUE
+reserved:	CONTROL
+reserved:	CONTROLS
+reserved:	CONVERSION		# not supported yet
+reserved:	CONVERTING
+reserved:	COPY
+reserved:	CORR
+reserved:	CORRESPONDING
+reserved:	COUNT
+reserved:	CURRENCY
+reserved:	DATA
+reserved:	DATE
+reserved:	DATE-COMPILED
+reserved:	DATE-WRITTEN
+reserved:	DAY
+reserved:	DAY-OF-WEEK
+reserved:	DE
+reserved:	DEBUGGING
+reserved:	DECIMAL-POINT
+reserved:	DECLARATIVES
+reserved:	DEFAULT			# COBOL 2002 != GCOS (unsupported)
+reserved:	DELETE
+reserved:	DELIMITED
+reserved:	DELIMITER
+reserved:	DEPENDING
+reserved:	DESCENDING
+reserved:	DESTINATION
+reserved:	DETAIL
+reserved:	DISABLE
+reserved:	DISPLAY
+reserved:	DIVIDE
+reserved:	DIVISION
+reserved:	DOWN
+reserved:	DUPLICATES
+reserved:	DYNAMIC
+reserved:	EBCDIC
+reserved:	EGI
+reserved:	ELSE
+reserved:	EMI
+reserved:	ENABLE
+reserved:	END
+reserved:	END-ADD
+reserved:	END-CALL
+reserved:	END-COMPUTE
+reserved:	END-DELETE
+reserved:	END-DIVIDE
+reserved:	END-EVALUATE
+reserved:	END-IF
+reserved:	END-MULTIPLY
+reserved:	END-OF-PAGE
+reserved:	END-PERFORM
+reserved:	END-READ
+reserved:	END-RECEIVE
+reserved:	END-RETURN
+reserved:	END-REWRITE
+reserved:	END-SEARCH
+reserved:	END-START
+reserved:	END-STRING
+reserved:	END-SUBTRACT
+reserved:	END-UNSTRING
+reserved:	END-WRITE
+reserved:	ENTER
+reserved:	ENVIRONMENT
+reserved:	EOP
+reserved:	EQUAL
+reserved:	EQUALS=EQUAL
+reserved:	ERROR
+reserved:	ESI
+reserved:	EVALUATE
+reserved:	EVERY
+reserved:	EXCEEDS=GREATER
+reserved:	EXCEPTION
+reserved:	EXIT
+reserved:	EXTEND
+reserved:	EXTERNAL
+reserved:	FALSE
+reserved:	FD
+reserved:	FILE
+reserved:	FILE-CONTROL
+reserved:	FILLER
+reserved:	FINAL
+reserved:	FIRST
+reserved:	FLOAT-LONG
```

Comment:
that doesn't match https://support.bull.com/ols/product/system/gcos7/gcos7-com/g7-dps7000/doc-com/docf/g/47X257TN23-mar2005/doc/05ula04.pdf A1 - do we currently need to have FLOAT-LONG in here to alias COMP-10? Otherwise this ad other COBOL2002 words should be removed.

Reply:
Indeed some of the "COBOL 2002" words are not in the list from the GCOS documentation, so I dropped them.

About `FLOAT-LONG` and the likes, however, the purpose of their appearance in the list is for the definition of aliases. For instance `FLOAT-LONG` is used as it matches the definition of `COMP-10`, and the definition of the alias `COMP-10=FLOAT-LONG` requires `FLOAT-LONG` to be included in the list. I can see where this may be an issue. Another alternative would be to add a dedicated entry for `COMP-10` and `COMP-15` in `cobc/reserved.c` as "GCOS extension"s. Would you prefer the latter approach? Or another I am currently failing to identify?

Reply:
For the latter, would an entry with
```c
  { "COMP-10",		0, 0, FLOAT_LONG,		/* GCOS extension */
				0, 0
  },
```
be enough?

Reply:
Adjusting reserved.c sounds good, your proposed change would be correct, you'd only need the alias shortform<->longform `COMPUTATIONAL` in gcos.words then and have the cobol2002-usages be `not-reserved:` in gcos.conf after enabling everything.
This has also the benefit that you'll likely add a syntax check with `-std=gcos` that compiles `01 FLOATS. 05 FLOAT-LONG USAGE COMP-10 VALUE 1.2456.  05 FLOAT-SHORT ....`.

Reply:
... and this has the benefit that `-std=default` will also compile those `USAGE`s.

Reply:
Ok I can try to do this. I'm not sure to exactly follow what you mean by: "then and have the cobol2002-usages be not-reserved: in gcos.conf after enabling everything.". 

The additional issue I have with the approach is that I need to define `COMP-1` to be equivalent to `BINARY-SHORT` without enabling the latter as a reserved word, yet `COMP-1` is already defined in `reserved.c`.

Reply:
I haven't checked this in detail, but are you sure, that this doesn't work?
If I do `cobc -std=rm-strict -freserved=BANANA:APPLE --list-reserved`  I get a  "BANANA not implemented", if I do `cobc -std=rm-strict -freserved=BANANA:FLOAT-LONG --list-reserved` I get a "BANANA implemented", which also should work that way.
If it works then it should also work to have `reserved: COMP-1=BINARY-SHORT COMPUTATIONAL-1=BINARY-SHORT`.
I meant to have the part

```
# not-reserved:
# Value: Word to be taken out of the reserved words list
# -fill upon request-"
```

after `include "lax.conf-inc"` (should be there in any case) then have `not-reserved: FLOAT-LONG` and friends - but on second thought: if not using the strict dialect, then there are word conflicts - that's the way it is, we can leave it that way.

Still wondering about "doesn't it work to just have `reserved: COMP-1=BINARY-SHORT`? Is there an error message, does it just "don't work"?

Reply:
> If it works then it should also work to have `reserved: COMP-1=BINARY-SHORT COMPUTATIONAL-1=BINARY-SHORT`.

Indeed I did not check the limitation myself before, and in the end you are right, it turns out we can approach the problem as suggested.